api/api.go

```diff
@@ -52,15 +65,28 @@ func createCORSHandler(allowedOrigins string, router *mux.Router) http.Handler {
 func routes(router *mux.Router, hc *healthcheck.HealthCheck) *RendererAPI {
 	api := RendererAPI{router: router}
 
-	handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
-		// Configure the "http.route" for the HTTP instrumentation.
-		handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
-		api.router.Handle(pattern, handler)
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(context.Background(), "error occurred when getting config", err)
+		return nil
+	}
+
+	if cfg.OtelEnabled {
+		handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
+			// Configure the "http.route" for the HTTP instrumentation.
+			handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+			api.router.Handle(pattern, handler)
+		}
+
+		handleFunc("/render/{render_type}", api.renderTable)
+		handleFunc("/parse/html", api.parseHTML)
+	} else {
+		api.router.HandleFunc("/render/{render_type}", api.renderTable).Methods("POST")
+		api.router.HandleFunc("/parse/html", api.parseHTML).Methods("POST")
```

Comment:
This seems like an error prone way of handling the routing, is there another approach which means we don't have two separate route definitions?

Reply:
This has been refactored, please let me know if it's better than the previous iteration :)